lib/src/idx/planner/iterators.rs

```diff
@@ -179,16 +193,62 @@ impl IndexRangeThingIterator {
 	}
 }
 
+pub(crate) struct IndexUnionThingIterator {
+	values_iter: IntoIter<(Vec<u8>, Vec<u8>)>,
+	current: Option<(Vec<u8>, Vec<u8>)>,
+}
+
+impl IndexUnionThingIterator {
+	pub(super) fn new(opt: &Options, ix: &DefineIndexStatement, a: &Array) -> Self {
+		// We create an iterator, as we are going iterate over every values of the array.
+		// Each item of the iterator are the prefix keys (begin and end).
+		let values: Vec<(Vec<u8>, Vec<u8>)> =
+			a.0.iter()
+				.map(|v| {
+					let a = Array::from(v.clone());
+					let beg = Index::prefix_ids_beg(opt.ns(), opt.db(), &ix.what, &ix.name, &a);
+					let end = Index::prefix_ids_end(opt.ns(), opt.db(), &ix.what, &ix.name, &a);
+					(beg, end)
+				})
+				.collect();
```

Comment:
Implementation question, but why are iterators passed around instead of collections like vectors? If there is a collect anyway, it's not lazy, is it?

Reply:
That's absolutely accurate! Changes:
- Replace intermediate Vec with VecDeque avoiding the need for an iterator.
- Simplify next_batch loop logic using 'while let'.